src/menus/NavigationMenus.cpp

```diff
@@ -29,57 +29,46 @@ void NextOrPrevFrame(AudacityProject &project, bool forward)
    auto temp1 = AButton::TemporarilyAllowFocus();
    auto temp2 = ASlider::TemporarilyAllowFocus();
    auto temp3 = MeterPanel::TemporarilyAllowFocus();
-
-   auto &toolManager = ToolManager::Get( project );
-   auto botDock = toolManager.GetBotDock();
-
-
-   // Define the set of windows we rotate among.
-   static const unsigned rotationSize = 3u;
-
-   wxWindow *const begin [rotationSize] = {
+   
+   std::vector<wxWindow*> seq = {
       ProjectWindow::Get( project ).GetTopPanel(),
       &TrackPanel::Get( project ),
-      botDock,
-   };
-
-   const auto end = begin + rotationSize;
-
-   // helper functions
-   auto IndexOf = [&](wxWindow *pWindow) {
-      return std::find(begin, end, pWindow) - begin;
+      ProjectWindow::Get(project).GetEffectsWindow(),
+      ToolManager::Get( project ).GetBotDock(),
    };
 
    auto FindAncestor = [&]() {
       wxWindow *pWindow = wxWindow::FindFocus();
-      unsigned index = rotationSize;
-      while ( pWindow &&
-              (rotationSize == (index = IndexOf(pWindow) ) ) )
+      unsigned index = seq.size();
+      while (pWindow)
+      {
+         auto it = std::find(seq.begin(), seq.end(), pWindow);
+         if(it != seq.end())
+            return static_cast<size_t>(std::distance(seq.begin(), it));
          pWindow = pWindow->GetParent();
-      return index;
+      }
+      return seq.size();
    };
 
    const auto idx = FindAncestor();
-   if (idx == rotationSize)
+   if (idx == seq.size())
       return;
 
    auto idx2 = idx;
-   auto increment = (forward ? 1 : rotationSize - 1);
-
-   while( idx != (idx2 = (idx2 + increment) % rotationSize) ) {
-      wxWindow *toFocus = begin[idx2];
-      bool bIsAnEmptyDock=false;
-      if( idx2 != 1 )
-         bIsAnEmptyDock = ((idx2==0) ? toolManager.GetTopDock() : botDock)->
-         GetChildren().GetCount() < 1;
+   const auto increment = (forward ? 1 : static_cast<int>(seq.size()) - 1);
```

Comment:
How about adding a pre-condition assert check before this line, an assert(seq.size() != 0)?  If seq.size() is 0 and forward is false then increment will become a very large number as size_t is an unsigned number.

Reply:
It can't be zero-sized, though it may contain nullptrs